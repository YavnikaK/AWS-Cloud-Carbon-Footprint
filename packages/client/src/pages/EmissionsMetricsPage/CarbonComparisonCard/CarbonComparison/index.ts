/*
 * Â© 2021 Thoughtworks, Inc.
 */

export { default } from './CarbonComparison'
