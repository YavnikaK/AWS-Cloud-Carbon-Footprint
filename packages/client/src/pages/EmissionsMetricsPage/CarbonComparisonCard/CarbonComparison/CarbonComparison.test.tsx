/*
 * Â© 2021 Thoughtworks, Inc.
 */

import React from 'react'
import { create } from 'react-test-renderer'
import CarbonComparison from './CarbonComparison'
import { fireEvent, render } from '@testing-library/react'
import { act } from 'react-dom/test-utils'

describe('CarbonComparison', () => {
  const FakeIcon = () => <div />
  const exampleComparisons = {
    optionOne: {
      icon: <FakeIcon />,
      total: 23,
      textOne: 'CO2e emissions from',
      textTwo: 'optionOne comparison',
      source: { href: '', title: 'optionOneSource' },
    },
    optionTwo: {
      icon: <FakeIcon />,
      total: 150,
      textOne: 'CO2e emissions from',
      textTwo: 'optionTwo comparison',
      source: { href: '', title: 'optionTwoSource' },
    },
  }

  const testProps = {
    formatNumber: jest.fn(),
    totalMetricTons: 100,
    comparisons: exampleComparisons,
    selection: 'optionOne',
    updateSelection: jest.fn(),
    updateButtonColor: jest.fn(),
  }

  it('renders with correct configuration and markup', () => {
    const root = create(<CarbonComparison {...testProps} />)

    expect(root.toJSON()).toMatchSnapshot()
  })

  it('calls the number formatter on the passed metric tons and comparison total', () => {
    render(<CarbonComparison {...testProps} />)

    expect(testProps.formatNumber).toHaveBeenCalledTimes(2)
    expect(testProps.formatNumber).toHaveBeenCalledWith(
      testProps.totalMetricTons,
      1,
    )
    expect(testProps.formatNumber).toHaveBeenCalledWith(
      testProps.comparisons.optionOne.total,
    )
  })

  it('calls the updateSelection function when an option is selected from the dropdown', () => {
    const { getByText } = render(<CarbonComparison {...testProps} />)

    const optionOneButton = getByText('optionOne')
    act(() => {
      fireEvent.click(optionOneButton)
    })

    expect(testProps.updateSelection).toHaveBeenCalledTimes(1)
    expect(testProps.updateSelection).toHaveBeenCalledWith('optionOne')
  })

  it('calls the updateButtonColor when rendering each option button', () => {
    render(<CarbonComparison {...testProps} />)

    expect(testProps.updateButtonColor).toHaveBeenCalledTimes(2)
    expect(testProps.updateButtonColor).toHaveBeenCalledWith('optionOne')
    expect(testProps.updateButtonColor).toHaveBeenCalledWith('optionTwo')
  })

  it('displays the selected comparison information and the rest of the prop data', () => {
    const { getByText, getByRole, getAllByRole } = render(
      <CarbonComparison {...testProps} />,
    )

    const selected = testProps.comparisons[testProps.selection]
    const buttons = getAllByRole('button')

    expect(getByText(selected.textOne)).toBeInTheDocument()
    expect(getByText(selected.textTwo)).toBeInTheDocument()
    expect(buttons).toHaveLength(2)
    expect(buttons[0]).toHaveTextContent('optionOne')
    expect(buttons[1]).toHaveTextContent('optionTwo')
    expect(getByRole('link')).toHaveTextContent(selected.source.title)
    expect(getByRole('link')).toHaveAttribute('href', selected.source.href)
  })
})
