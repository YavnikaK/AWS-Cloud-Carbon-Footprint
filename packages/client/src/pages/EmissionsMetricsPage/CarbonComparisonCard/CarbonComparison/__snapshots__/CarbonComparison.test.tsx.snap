// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarbonComparison renders with correct configuration and markup 1`] = `
<div>
  <div
    className="MuiCardContent-root makeStyles-topContainer-4"
  >
    <p
      className="MuiTypography-root makeStyles-title-1 MuiTypography-body1 MuiTypography-gutterBottom"
    >
      Your cumulative emissions are
    </p>
    <p
      className="MuiTypography-root makeStyles-metricOne-7 MuiTypography-h4"
      data-testid="co2"
      id="metric-one"
    >
       metric tons CO2e
    </p>
    <p
      className="MuiTypography-root makeStyles-posOne-2 MuiTypography-body1"
    >
      that is equivalent to
    </p>
  </div>
  <div
    className="MuiCardContent-root makeStyles-bottomContainer-5"
  >
    <div
      className="MuiCardContent-root"
    >
      <div />
    </div>
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root makeStyles-posTwo-3 MuiTypography-h5"
      >
        CO2e emissions from
      </p>
      <p
        className="MuiTypography-root makeStyles-metricTwo-8 MuiTypography-h3"
        data-testid="comparison"
      />
      <p
        className="MuiTypography-root makeStyles-posTwo-3 MuiTypography-h5"
      >
        optionOne comparison
      </p>
    </div>
  </div>
  <div
    className="MuiCardActions-root makeStyles-buttonContainer-6 MuiCardActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained"
      disabled={false}
      id="optionOne"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        optionOne
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained"
      disabled={false}
      id="optionTwo"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        optionTwo
      </span>
    </button>
  </div>
  <p
    className="MuiTypography-root makeStyles-source-10 MuiTypography-body1"
    data-testid="epa-source"
  >
    Source:
     
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-sourceLink-11 MuiTypography-colorPrimary"
      href=""
      onBlur={[Function]}
      onFocus={[Function]}
      rel="noopener"
      target="_blank"
    >
      optionOneSource
       
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root makeStyles-openIcon-12 MuiSvgIcon-fontSizeSmall"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
        />
      </svg>
    </a>
  </p>
</div>
`;
